libs/shared/flagd-core/src/lib/flagd-core.ts

```diff
@@ -0,0 +1,122 @@
+import {MemoryStorage, Storage} from './storage';
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValue,
+  GeneralError,
+  JsonValue,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+} from '@openfeature/server-sdk';
+
+/**
+ * Expose flag configuration setter and flag resolving methods.
+ */
+export class FlagdCore {
+  private _storage: Storage;
+
+  /**
+   * Optionally construct with your own storage layer.
+   */
+  constructor(storage?: Storage) {
+    this._storage = storage ? storage : new MemoryStorage();
+  }
+
+  /**
+   * Add flag configurations to the storage.
+   */
+  setConfigurations(cfg: string): void {
+    this._storage.setConfigurations(cfg);
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    evalCtx: EvaluationContext,
+  ): ResolutionDetails<boolean> {
+    return this.resolve(flagKey, defaultValue, evalCtx, (v) => {
+      return typeof v == 'boolean';
+    });
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    evalCtx: EvaluationContext,
+  ): ResolutionDetails<string> {
+    return this.resolve(flagKey, defaultValue, evalCtx, (v) => {
+      return typeof v == 'string';
+    });
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    evalCtx: EvaluationContext,
+  ): ResolutionDetails<number> {
+    return this.resolve(flagKey, defaultValue, evalCtx, (v) => {
+      return typeof v == 'number';
+    });
+  }
+
+  resolveObjectEvaluation<T extends JsonValue>(
+    flagKey: string,
+    defaultValue: T,
+    evalCtx: EvaluationContext,
+  ): ResolutionDetails<T> {
+    return this.resolve(flagKey, defaultValue, evalCtx, (v) => {
+      return typeof v == 'object';
+    });
+  }
+
+  private resolve<T extends FlagValue>(
+    flagKey: string,
+    defaultValue: T,
+    evalCtx: EvaluationContext,
+    guard: (a: FlagValue) => boolean,
+  ): ResolutionDetails<T> {
+
+    // flag exist check
+    const flag = this._storage.getFlag(flagKey);
+    if (flag === undefined) {
+      throw new FlagNotFoundError(`flag: ${flagKey} not found`);
+    }
+
+    // flag status check
+    if (flag.state === 'DISABLED') {
+      throw new GeneralError(`flag: ${flagKey} is disabled`);
```

Comment:
IMO, disabled is not an error - we expect disabled flags to be evaluated, so an exception based control flow is an anti-pattern. This should just return the default with the `Reason` = `DISABLED`.

Reply:
Good point, I will update this